### ISSUE TITLE
Unnecessary wider permissions required for consistent API intent

### ISSUE THREAD

**Issue by enxebre** [kubernetes-sigs/cluster-api-provider-aws#4342]
/kind bug

**What steps did you take and what happened:**
I'm expressing intent in the API similar to
```
spec:
  additionalSecurityGroups:
  - id: sg-0e42acdc7c8b1f9a1
  additionalTags:
    expirationDate: 2023-06-14T14:51+00:00
    kubernetes.io/cluster/example-vmtl4: owned
  ami:
    id: ami-01735c4f217bf0ae4
  cloudInit:
    insecureSkipSecretsManager: true
    secureSecretsBackend: secrets-manager
  iamInstanceProfile: example-vmtl4-worker
  instanceType: m5.large
  rootVolume:
    size: 64
    type: gp3
  subnet:
    id: subnet-0d10940bdfb2aeb29
  uncompressedUserData: true
```

No InstanceMetadataOptions.

Now when updating to latest version of the controller I got a requirement for wider permissions even though I'm not exercising them.

```
 - lastTransitionTime: "2023-06-14T11:25:42Z"
    message: "failed to create AWSMachine instance: UnauthorizedOperation: You are
      not authorized to perform this operation.\n\tstatus code: 403, request id: 8b50824a-16ab-40cf-b737-47cb80ad34e4"
    reason: InstanceProvisionFailed
    severity: Error
    status: "False"
    type: InstanceReady
```

The field is optional and my API intent is the same than it was before, and so I consider the controller requiring any additional permissions to operate a disruptive breaking change

I suspect this is related to the instance metadata changes github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4037
and the fact we are setting defaults in the backend `infrav1.SetDefaults_AWSMachineSpec(&awsMachine.Spec)`  and so bypassing API intent.
I'd expect `SetDefaults_AWSMachineSpec` to honour nil API input as `getInstanceMetadataOptionsRequest` does. Then
seems to me that the in [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4037/files#r1129064749) should be fixed by accounting for nil derefs in `ensureInstanceMetadataOptions`

**What did you expect to happen:**
New version of controller do not require wider permissions for same API intent.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by muraee** [comment]
the issue was a missing `ec2:DescribeInstanceTypes` permission introduced here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4054

logs show this as a warning, but it could be signaled better in the condition message
```
I0616 08:57:20.438841       1 recorder.go:103] "events: Operation DescribeInstanceTypes failed with a credentials or permission issue" type="Warning" object={"kind":"AWSCluster","namespace":"clusters","name":"mraee-test03","uid":"d69fdbb2-3f2b-4fac-a436-dd48b6d1a5ed","apiVersion":"infrastructure.cluster.x-k8s.io/v1beta2","resourceVersion":"3018367"} reason="UnauthorizedOperation"
```

**Comment by enxebre** [comment]
I remain on the opinion that requiring wider perms when the API intent didn't change is bug so I think we should try to fix that as well to not force widen up IAM unnecessarily.

**Comment by damdo** [comment]
I'm also seeing
```
failed to create AWSMachine instance: UnauthorizedOperation: You are not authorized to perform this operation
```
on `v2.1.x`.

**Comment by damdo** [comment]
Any thoughts  @Ankitasw @Skarlso ? Thanks

**Comment by damdo** [comment]
@enxebre do you think the merging of #4347 grants us closing this one now?

**Comment by Ankitasw** [comment]
> I'm also seeing

>failed to create AWSMachine instance: UnauthorizedOperation: You are not authorized to perform this operation
on v2.1.x.

Is this issue resolved or still persists? In case it persists, we might have to revert [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4347) change. cc @richardcase @enxebre

**Comment by damdo** [comment]
Hey @Ankitasw, yes the issue was resolved for me by #4347

**Comment by Ankitasw** [comment]
ok then its good to close.
